docs/agent/secrets.md

```diff
@@ -150,16 +159,70 @@ Regardless of the workflow, the user should **take great care to secure the
 executable itself**, including setting appropriate permissions and considering
 the security implications of their executable in their environment.
 
-In particular, the executable **MUST** (the agent will refuse to use it otherwise):
+### Agent security requirements
+
+The agent will run `secret_backend_command` executable as a sub-process. The
+execution pattern differ on Linux and Windows.
 
-- Belong to the same user running the agent (usually `dd-agent`).
+#### Linux
+
+On Linux, the executable set as `secret_backend_command` **MUST** (the agent
+will refuse to use it otherwise):
+
+- Belong to the same user running the agent (by default `dd-agent` or `root`
+  inside a container).
 - Have **no** rights for `group` or `other`.
 - Have at least `exec` right for the owner.
+
+Also:
+- The executable will not share any environment variables with the agent.
+- Never output sensitive information on STDERR. If the binary exit with a
+  different status code than `0` the agent will log the standard error output
+  of the executable to ease troubleshooting.
+
+#### Windows
+
+When installing the agent (or upgrading from a version without the secrets
+feature) you MUST add the `SECRETUSER=true` flag to activate the feature. This
+will create the right user to fetch secrets (see below). You no longer need to
+add that flag when upgrading the agent.
+
+ex:
+```
+msiexec /qn /i datadog-agent-6-latest.amd64.msi SECRETUSER=true
+```
+
+On windows, a specific user named `datadog_secretuser` is created by the
```

Comment:
_Suggestion_: let's make the username unpredictable. This might add some security, if attackers do not know in advance. We can keep the username in the registry, too. Let me clarify: we do _NOT_ have to do this, it's just a thought.